### ISSUE TITLE
pm2 start and never return

### ISSUE THREAD

**Issue by bangbang93** [Unitech/pm2#1764]
I use json declare my application 
```javascript
{
  "apps" : [{
    "name"        : "push",
    "script"      : "./index.js",
    "watch"       : true,
    "merge_logs"  : true,
    "env": {
      "DEBUG": "cofactories-push:*",
      "NODE_ENV": "production"
    }
  }]
}
```
then I use "pm2 start process.json" to start my application .But it just stop here 
![image](https://cloud.githubusercontent.com/assets/3430784/11138413/ec3898ac-89fc-11e5-9ad7-a10434eb1df9.png)
And pm2 list can't find it .
But if I use "pm2 start index.js", it works 
pm2 0.15.10
node 4.2.2

**Comment by bangbang93** [comment]
when i use "pm2 restart push --watch",daemon even stop running(I use another ssh terminal run "pm2 list",got "spawing pm2 daemon")and lost my process list
```
2015-11-13 21:56:43: App name:www id:12 exited with code SIGINT
2015-11-13 21:56:43: Process with pid 8592 killed
2015-11-13 21:56:43: Starting execution sequence in -fork mode- for app name:www id:12
2015-11-13 21:56:43: App name:www id:12 online
2015-11-13 22:02:47: [PM2] Error caught by domain:
TypeError: God.watch._watchers[pm2_env.name].close is not a function
    at Object.ClusterMode.God.watch.enable (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/lib/Watcher.js:20:41)
    at God.restartWatch (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/lib/God/ActionMethods.js:576:19)
    at Server.onmessage (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/node_modules/pm2-axon-rpc/lib/server.js:105:6)
    at emitTwo (events.js:87:13)
    at RepSocket.emit (events.js:172:7)
    at Parser.<anonymous> (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/node_modules/pm2-axon/lib/sockets/rep.js:51:15)
    at emitOne (events.js:77:13)
    at Parser.emit (events.js:169:7)
    at Parser._write (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/node_modules/pm2-axon/node_modules/amp/lib/stream.js:91:16)
    at doWrite (_stream_writable.js:292:12)
    at writeOrBuffer (_stream_writable.js:278:5)
2015-11-13 22:02:47: [PM2] Trying to update PM2...
Be sure to have the latest version by doing `npm install pm2@latest -g` before doing this procedure.
[PM2] Stopping PM2...
2015-11-13 22:02:48: [Watch] Stop watching 0
2015-11-13 22:02:48: Stopping app:proxy id:0
2015-11-13 22:02:48: App name:proxy id:0 disconnected
2015-11-13 22:02:48: App name:proxy id:0 exited with code SIGTERM
2015-11-13 22:02:48: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:48: Process with pid 25171 killed
[PM2] deleteProcessId process id 0
2015-11-13 22:02:48: Stopping app:protome id:1
2015-11-13 22:02:48: App name:protome id:1 exited with code SIGTERM
2015-11-13 22:02:48: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:48: Process with pid 21927 killed
[PM2] deleteProcessId process id 1
2015-11-13 22:02:48: Stopping app:customer id:2
2015-11-13 22:02:48: App name:customer id:2 exited with code SIGTERM
2015-11-13 22:02:48: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:48: Process with pid 8158 killed
[PM2] deleteProcessId process id 2
2015-11-13 22:02:49: Stopping app:preview id:3
2015-11-13 22:02:49: App name:preview id:3 exited with code SIGTERM
2015-11-13 22:02:49: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:49: Process with pid 21940 killed
[PM2] deleteProcessId process id 3
2015-11-13 22:02:49: Stopping app:admin id:4
2015-11-13 22:02:49: App name:admin id:4 exited with code SIGTERM
2015-11-13 22:02:49: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:49: Process with pid 21950 killed
[PM2] deleteProcessId process id 4
2015-11-13 22:02:49: Stopping app:control id:5
2015-11-13 22:02:49: App name:control id:5 exited with code SIGTERM
2015-11-13 22:02:49: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:49: Process with pid 21961 killed
[PM2] deleteProcessId process id 5
2015-11-13 22:02:49: Stopping app:activity id:6
2015-11-13 22:02:49: App name:activity id:6 exited with code SIGTERM
2015-11-13 22:02:49: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:49: Process with pid 21974 killed
[PM2] deleteProcessId process id 6
2015-11-13 22:02:49: [Watch] Stop watching 7
2015-11-13 22:02:49: Stopping app:auth id:7
2015-11-13 22:02:49: App name:auth id:7 exited with code SIGTERM
2015-11-13 22:02:49: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:49: Process with pid 21983 killed
[PM2] deleteProcessId process id 7
2015-11-13 22:02:49: Stopping app:pm2-http-interface id:8
2015-11-13 22:02:49: App name:pm2-http-interface id:8 exited with code SIGTERM
2015-11-13 22:02:49: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:49: Process with pid 21997 killed
[PM2] deleteProcessId process id 8
2015-11-13 22:02:49: [Watch] Stop watching 9
2015-11-13 22:02:49: Stopping app:cofactories id:9
2015-11-13 22:02:49: App name:cofactories id:9 disconnected
2015-11-13 22:02:49: App name:cofactories id:9 exited with code SIGTERM
2015-11-13 22:02:49: [HandleExit] PM2 is being killed, stopping restart procedure...
2015-11-13 22:02:50: Process with pid 22003 killed
[PM2] deleteProcessId process id 9
2015-11-13 22:02:50: [Watch] Stop watching 11
events.js:141
      throw er; // Unhandled 'error' event
      ^

TypeError: God.watch._watchers[pm2_env.name].close is not a function
    at Object.ClusterMode.God.watch.disable (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/lib/Watcher.js:102:41)
    at God.stopWatch (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/lib/God/ActionMethods.js:550:19)
    at Server.onmessage (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/node_modules/pm2-axon-rpc/lib/server.js:105:6)
    at emitTwo (events.js:87:13)
    at RepSocket.emit (events.js:172:7)
    at Parser.<anonymous> (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/node_modules/pm2-axon/lib/sockets/rep.js:51:15)
    at emitOne (events.js:77:13)
    at Parser.emit (events.js:169:7)
    at Parser._write (/home/cofactories/.nvm/versions/node/v4.2.2/lib/node_modules/pm2/node_modules/pm2-axon/node_modules/amp/lib/stream.js:91:16)
    at doWrite (_stream_writable.js:292:12)
    at writeOrBuffer (_stream_writable.js:278:5)
2015-11-13 22:04:23: [PM2][WORKER] Started with refreshing interval: 30000
2015-11-13 22:04:23: [[[[ PM2/God daemon launched ]]]]
2015-11-13 22:04:23: BUS system [READY] on port /home/cofactories/.pm2/pub.sock
2015-11-13 22:04:23: RPC interface [READY] on port /home/cofactories/.pm2/rpc.sock
2015-11-13 22:05:13: Starting execution sequence in -cluster mode- for app name:proxy id:0
2015-11-13 22:05:14: App name:proxy id:0 online
2015-11-13 22:05:14: Starting execution sequence in -fork mode- for app name:protome id:1
2015-11-13 22:05:14: App name:protome id:1 online
2015-11-13 22:05:14: Starting execution sequence in -fork mode- for app name:customer id:2
2015-11-13 22:05:14: App name:customer id:2 online
2015-11-13 22:05:14: Starting execution sequence in -fork mode- for app name:preview id:3
2015-11-13 22:05:14: App name:preview id:3 online
2015-11-13 22:05:14: Starting execution sequence in -fork mode- for app name:admin id:4
2015-11-13 22:05:14: App name:admin id:4 online
2015-11-13 22:05:14: Starting execution sequence in -fork mode- for app name:control id:5
2015-11-13 22:05:14: App name:control id:5 online
2015-11-13 22:05:14: Starting execution sequence in -fork mode- for app name:activity id:6
2015-11-13 22:05:14: App name:activity id:6 online
2015-11-13 22:05:14: Starting execution sequence in -fork mode- for app name:auth id:7
2015-11-13 22:05:14: App name:auth id:7 online
2015-11-13 22:05:14: Starting execution sequence in -fork mode- for app name:pm2-http-interface id:8
2015-11-13 22:05:15: App name:pm2-http-interface id:8 online
2015-11-13 22:05:15: Starting execution sequence in -cluster mode- for app name:cofactories id:9
2015-11-13 22:05:16: App name:cofactories id:9 online
2015-11-13 22:05:16: Starting execution sequence in -fork mode- for app name:push id:10
2015-11-13 22:05:16: App name:push id:10 online
2015-11-13 22:05:16: Starting execution sequence in -fork mode- for app name:www id:11
2015-11-13 22:05:16: App name:www id:11 online
```

**Comment by bangbang93** [comment]
When I edit my process.json
```
{
  "apps" : [{
    "name"        : "push",
    "script"      : "./index.js",
    "watch"       : false,
    "merge_logs"  : true,
    "env": {
      "DEBUG": "cofactories-push:*",
      "NODE_ENV": "production"
    }
  }]
}
```
stop watching directory ,pm2 can start my json ,but if I restart the process and start watching ,God daemon crashed

**Comment by likol** [comment]
I declare app name `app` I can start , but when I run `pm2 delete push` or `pm2 kill`, pm2 daemon will crash

part of my .json file

```
{
      "name": "push",
      "script": "./server.js",
      "log_date_format": "YYYY-MM-DD HH:mm Z",
      "instances": 1,
      //or 0 => 'max'
      "min_uptime": "200s",
      // 200 seconds, defaults to 1000
      "max_restarts": 15,
      // defaults to 15
      "max_memory_restart": "200M",
      // 1 megabytes, e.g.: "2G", "10M", "100K", 1024 the default unit is byte.
      "cron_restart": "1 0 * * *",
      "watch": false,
      "ignore_watch": [
        "[\\/\\\\]\\./",
        "node_modules"
      ],
      "merge_logs": true,
      "exec_interpreter": "node",
      "exec_mode": "cluster",
      "autorestart": false,
      // enable/disable automatic restart when an app crashes or exits
      "vizion": false,
      // enable/disable vizion features (versioning control)
      "env": {
        "NODE_ENV": "development",
        "PORT": 8883
      }
    },
```

**Comment by Unitech** [comment]
Well internally in the PM2 daemon, by calling your application "push" it overrides the .push function of the Array object ^^

**Comment by Unitech** [comment]
Just rename your application with another name than *push*

**Comment by kjhangiani** [comment]
This is kind of a ridiculous fix, I can't create apps that include the word "push" anywhere in them?  Seems excessive.. I have been trying to debug an opsworks failure, and I just came across this error.  I tried to create an app with the name "soxhub-api-nspusherqa"  note the word push being in the middle of the string... why would that be forbidden?  I can understand checking for an app named explicitly "push" or even with the prefix "push", but why can't I use it in the middle of the app name?
